program-runtime/src/accounts_data_meter.rs

```diff
@@ -35,36 +40,47 @@ impl AccountsDataMeter {
         self.maximum
     }
 
+    /// Return the initial amount of accounts data space used (in bytes)
+    pub fn initial(&self) -> u64 {
+        self.initial
+    }
+
+    /// Return the amount of accounts data space that has changed (in bytes)
+    pub fn delta(&self) -> i64 {
+        self.delta
+    }
+
     /// Return the current amount of accounts data space used (in bytes)
     pub fn current(&self) -> u64 {
-        self.current
+        /// NOTE: Mixed integer ops currently not stable, so copying the impl.
+        /// * https://github.com/rust-lang/rust/issues/87840
+        /// * https://github.com/a1phyr/rust/blob/47edde1086412b36e9efd6098b191ec15a2a760a/library/core/src/num/uint_macros.rs#L1039-L1048
+        const fn saturating_add_signed(lhs: u64, rhs: i64) -> u64 {
+            let (res, overflow) = lhs.overflowing_add(rhs as u64);
+            if overflow == (rhs < 0) {
+                res
+            } else if overflow {
+                u64::MAX
+            } else {
+                u64::MIN
+            }
+        }
+        saturating_add_signed(self.initial, self.delta)
     }
 
     /// Get the remaining amount of accounts data space (in bytes)
     pub fn remaining(&self) -> u64 {
-        self.maximum.saturating_sub(self.current)
+        self.maximum.saturating_sub(self.current())
     }
 
     /// Consume accounts data space, in bytes.  If `amount` is positive, we are *increasing* the
     /// amount of accounts data space used.  If `amount` is negative, we are *decreasing* the
     /// amount of accounts data space used.
     pub fn consume(&mut self, amount: i64) -> Result<(), InstructionError> {
-        if amount == 0 {
```

Comment:
nit: I found the name of this function and it's description like reading double negatives.  Maybe `consume` is not the best name for this function because we are not just using up a resource, but also replenishing it.  Maybe rename to `adjust_delta`, `adjust_remaining` or something along those lines?

Reply:
Yah, I definitely agree with you. Since this function is not new, I will address this in a subsequent PR (and so I don't need to have CI run again on this PR). Let me know if that's not ok.

Reply:
Tracking this here: #23022